vixos/ssh.py

```diff
@@ -23,20 +30,44 @@ def pubkey_text(self) -> str:
         return self.privkey.publickey().exportKey("OpenSSH").decode()
 
     def interactive_session(self, user: str, host: str) -> None:
-        # TODO: use a hardcoded known host key here instead?
-        # TODO: use paramiko session instead of shelling to ssh?
-        subprocess.check_call(
-            [
-                "ssh",
-                f"user@{host}",
-                "-i",
-                str(self.privkey_path),
-                "-o",
-                "StrictHostKeyChecking=no",
-                "-o",
-                "UserKnownHostsFile=/dev/null",
-            ]
-        )
+        ssh = self.ssh_session(user, host)
+        width, height = os.get_terminal_size()
+        shell_chan = ssh.invoke_shell(os.getenv('TERM', 'vt100'), width, height)
+
+        def signal_handler():
+            width, height = os.get_terminal_size()
+            shell_chan.resize_pty(width, height)
+        signal.signal(signal.SIGWINCH, signal_handler)
+
+        tty_attr = termios.tcgetattr(sys.stdin)
+        try:
+            # make terminal raw, disable line buffering
+            tty.setraw(sys.stdin.fileno())
+            tty.setcbreak(sys.stdin.fileno())
+            shell_chan.settimeout(0.0)
+
+            while True:
+                read_ready, _, _ = select.select([shell_chan, sys.stdin], [], [])
+                if shell_chan in read_ready:
+                    try:
+                        b = shell_chan.recv(1024)
+                        if len(b) == 0:
+                            print(f'\nConnection to {host} closed')
+                            break
+                        sys.stdout.write(b)
```

Comment:
```suggestion
                        sys.stdout.buffer.write(b)
```
(to avoid encoding errors)